### DISCUSSION TITLE
What is `{}` type?

### DISCUSSION THREAD

**Discussion by mipopon** [honojs/hono#2222] [Q&A] [answered]
I recently encounter this `{}` type a lot in the hono codebase. Also on honoX Readme it suggestes to [`import type {} from 'hono`](https://github.com/honojs/honox/blob/main/README.md#L173:~:text=import%20type%20%7B%7D%20from%20%27hono%27)

When I develop on my projects eslint shows me the error `Don't use `{}` as a type. `{}` actually means "any non-nullish value"`. 
Can anyone help why it's used and what the purpose is?

**Comment by yusukebe** [comment, marked as answer]
@mipopon 

We want to import just `types` from `hono` in that `global.d.ts`, but it can't. So, I wrote it. But if you have any ideas, let us know.

**Comment by mipopon** [comment]
@yusukebe I see `import type {} from 'hono';` is a way to import all types. I wasn't aware of that, thank you for the clarification.

So then this `{}` is used as`any non-nullish value`?
```
export type BlankSchema = {};
export type BlankInput = {};
```

**Reply by yusukebe** [reply]
@mipopon 

I want to use it as a "blank record".

Do you think we should not use `{}`?

**Reply by mipopon** [reply]
Initially I was suggesting to use more narrow type for such a default value like `Record<string, unknown>` but as a default value like this
`S extends Schema = BlankSchema,`  then BlankSchema is just the most wide default to avoid having unnecessary type issues while Schema will ensure type safety I assume? I don't have a specific case where the wide default value would cause issues.

 It was just a curiosity driven question. Sorry for bothering you and thank you for answering.
